### PR TITLE
Clarify how regex comparison operators and full text search can be used on the same field

### PR DESCRIPTION
**Pull request by hlcianfagna**
## Summary of the changes / Why this improves CrateDB

Closes: #14572

## Checklist

 - [X] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [X] This does not contain breaking changes

### CODE REVIEW

File: docs/general/dql/selects.rst [hunk 1]
```diff
@@ -210,7 +210,25 @@ It returns ``true`` if the string matches the pattern, ``false`` if not, and
...
+
+Example::
+
+    cr> CREATE TABLE tbl1(
+    ...   a TEXT,
+    ...   INDEX a_ft USING FULLTEXT(a)
+    ... );
+    CREATE OK, 1 row affected (... sec)
+
+::
```
Comment (matriv) [inline docs/general/dql/selects.rst:229]: This is not needed, this causes the error in build.

File: docs/general/dql/selects.rst [hunk 2]
```diff
@@ -210,7 +210,25 @@ It returns ``true`` if the string matches the pattern, ``false`` if not, and
 To negate the matching, use the optional ``!`` prefix. The operator returns
 ``true`` if the string does not match the pattern, ``false`` otherwise.
 
-The regular expression pattern is implicitly anchored, that means that the
+To make the search case insensitive use the ``*`` suffix.
```
Comment (matriv) [inline docs/general/dql/selects.rst:213]: maybe it's more clear to just say:
```suggestion
To make the search case insensitive use  ``~*`` and ``!~*``
```

File: docs/general/dql/selects.rst [hunk 3]
```diff
@@ -210,7 +210,25 @@ It returns ``true`` if the string matches the pattern, ``false`` if not, and
...
+
+These operators are not intended to be used against
```
[the remark this answers is not in the document]
Comment (hlcianfagna) [reply docs/general/dql/selects.rst]: It seems I cannot have an example nested into a note box, do you know of any page where we may have done this already that I can take as reference?

[the remark this answers is not in the document]
Comment (matriv) [reply docs/general/dql/selects.rst]: https://github.com/crate/crate/blob/master/docs/sql/general/constraints.rst?plain=1#L131

You need indent the example further 2 cols from the 4 cols you indent the NOTE content.

[the remark this answers is not in the document]
Comment (matriv) [reply docs/general/dql/selects.rst]: btw you can use: https://github.com/crate/crate/blob/master/devs/docs/write_docs.rst#run-the-sphinx-dev-server to see your changes locally and fast without rebuilding all the docs.

**Review by matriv — APPROVED** [review]
thank you!

**Comment by matriv** [comment]
retest this please

**Pull request merged**